sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/RegionCannotChange.java

```diff
@@ -21,17 +21,10 @@
         }
 
         List<ConfigValidationError> errors = new ArrayList<>();
-        boolean regionWasAdded = oldConfig.isPresent() && !oldConfig.get().getRegion().isPresent() &&
-                newConfig.getRegion().isPresent();
-        boolean regionWasUnset = oldConfig.isPresent() && oldConfig.get().getRegion().isPresent() &&
-                !newConfig.getRegion().isPresent();
-        boolean regionWasChanged = oldConfig.isPresent() && oldConfig.get().getRegion().isPresent() &&
-                newConfig.getRegion().isPresent() &&
-                !newConfig.getRegion().get().equals(oldConfig.get().getRegion().get());
-        if (regionWasAdded || regionWasUnset || regionWasChanged) {
+        if (!oldConfig.get().getRegion().equals(newConfig.getRegion())) {
             errors.add(ConfigValidationError.transitionError(
                     "region",
-                    oldConfig.get().getUser(),
+                    oldConfig.get().getRegion().toString(),
                     newConfig.getUser(),
                     "Region for old service must remain the same across deployments."
```

Comment:
What does "old" service mean in this context?

Reply:
It means the `ServiceSpec` generated from the last config target, if any. That's the name that's used in the interface definition.